source/ingress.go

```diff
@@ -232,6 +232,15 @@ func endpointsFromIngress(ing *v1beta1.Ingress) []*endpoint.Endpoint {
 		}
 		endpoints = append(endpoints, endpointsForHostname(rule.Host, targets, ttl)...)
 	}
+
+	if ing.Annotations[hostnameAnnotationKey] != "" {
+		hostnames := strings.Split(ing.Annotations[hostnameAnnotationKey], ",")
```

Comment:
the logic differs from https://github.com/kubernetes-incubator/external-dns/blob/68be609648b43dc067926979891bf5eb468b0dd3/source/service.go#L189, we should make it consistent (e.g. the code in `service.go` removes whitespace anywhere

Maybe extract into a new function and use it in both places?